pyanaconda/modules/common/structures/payload.py

```diff
@@ -0,0 +1,205 @@
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from dasbus.structure import DBusData
+from dasbus.typing import *  # pylint: disable=wildcard-import
+from pyanaconda.core.constants import URL_TYPE_BASEURL
+
+__all__ = ["URLData", "SSLConfigurationData", "PackagingConstrainsData"]
+
+
+class URLData(DBusData):
+    """Structure which contains link to URL and type of URL."""
+
+    def __init__(self):
+        self._url = ""
+        self._type = URL_TYPE_BASEURL
+
+    @property
+    def url(self) -> Str:
+        """Get link to the repository."""
+        return self._url
+
+    @url.setter
+    def url(self, url: Str):
+        """Set link to the repository."""
+        self._url = url
+
+    @property
+    def type(self) -> Str:
+        """Get type of the url.
+
+        Possible values:
+        - BASEURL
+        - MIRRORLIST
+        - METALINK
+        """
+        return self._type
+
+    @type.setter
+    def type(self, url_type: Str):
+        """Set type of the url.
+
+        For list of possible values see type getter.
+        """
+        self._type = url_type
+
+
+class SSLConfigurationData(DBusData):
+    """Structure with SSL configuration settings."""
+
+    def __init__(self):
+        self._ca_cert_path = ""
+        self._client_cert_path = ""
+        self._client_key_path = ""
+
+    @property
+    def ca_cert_path(self) -> Str:
+        """Get CA certificate path."""
+        return self._ca_cert_path
+
+    @ca_cert_path.setter
+    def ca_cert_path(self, ca_cert_path: Str):
+        """Set CA certificate path."""
+        self._ca_cert_path = ca_cert_path
+
+    @property
+    def client_cert_path(self) -> Str:
+        """Get client certificate path."""
+        return self._client_cert_path
+
+    @client_cert_path.setter
+    def client_cert_path(self, client_cert_path: Str):
+        """Set client certificate path."""
+        self._client_cert_path = client_cert_path
+
+    @property
+    def client_key_path(self) -> Str:
+        """Get client key path."""
+        return self._client_key_path
+
+    @client_key_path.setter
+    def client_key_path(self, client_key_path: Str):
+        """Set client key path."""
+        self._client_key_path = client_key_path
+
+
+class RepoCostData(DBusData):
+    """Structure to store cost value of the repository with flag if this should be used."""
+
+    def __init__(self):
+        self._cost = 0
+        self._enabled = False
+
+    def set_cost(self, cost):
+        """Set cost value of this repository and set it enabled.
+
+        This is a shortcut to avoid calling enable = True all the time.
+
+        :param cost: cost value of this repository
+        :type cost: int
+        """
+        self.cost = cost
+        self.enabled = True
+
+    @property
+    def cost(self) -> Int:
```

Comment:
The default value of the cost is 1000, so I would use the default value and avoid this DBus structure. See: https://dnf.readthedocs.io/en/latest/conf_ref.html#repo-options

You can still remember whether you are using a default or a user value internally in the module for the kickstart and dnf outputs.

Reply:
Fixed. Drawback of this is to that we have to have private API for payload to know that this value has changed.